### PR TITLE
Openstack inventory updates from Tower.

### PR DESCRIPTION
**Pull request by cchurch**
Includes changes from https://github.com/ansible/ansible/pull/10491 and https://github.com/ansible/ansible/pull/10776 and allows specifying a config file via environment variable.

### CODE REVIEW

File: plugins/inventory/openstack.py [hunk 1]
```diff
@@ -152,8 +156,7 @@ def main():
         elif args.host:
             inventory.get_host(args.host)
     except shade.OpenStackCloudException as e:
-        print(e.message)
-        sys.exit(1)
+        sys.exit(e.message)
```
Comment (abadger) [inline plugins/inventory/openstack.py]: I'd rather be explicit and use:

``` python
sys.stderr.write('%s\n' % e.message)
sys.exit(1)
```

**Comment by abadger** [comment]
@emonty Care to review this and +1 if it looks good to merge?

File: plugins/inventory/openstack.py (hunk 1 above)
Comment (cchurch) [inline plugins/inventory/openstack.py]: Pushed a new commit to fix.  Looking at the other changes in https://github.com/ansible/ansible/pull/10776, that PR is really inconsistent in this regard.

File: plugins/inventory/openstack.py [hunk 2]
```diff
@@ -51,11 +51,14 @@
 class OpenStackInventory(object):
 
     def __init__(self, private=False, refresh=False):
+        config_files = os_client_config.config.CONFIG_FILES
+        if os.environ.get('OPENSTACK_CONFIG_FILE', None):
+            config_files.insert(0, os.environ['OPENSTACK_CONFIG_FILE'])
```
Comment (emonty) [inline plugins/inventory/openstack.py]: Would you be amenable to changing this to "OS_CLIENT_CONFIG_FILE"? If so, I'd like to upstream support for setting this into os-client-config itself (it seems like a generally useful feature) We can land it here safely (with the name change) until we get it landed there.

**Comment by emonty** [comment]
I created a commit for supporting the env-var oerride. in openstack/os-client-config here: https://review.openstack.org/198502.

**Comment by emonty** [comment]
@cchurch you ok with the env var name change? We've landed the support for it generally in os-client-config and I can cut a release asap if you're good with that approach.

**Comment by cchurch** [comment]
@emonty I've removed my environment variable override since it's now supported by os-client-config.

@abadger @bcoca I've rebased & squashed, should be ready to go.

**Pull request merged**